corehq/apps/receiverwrapper/util.py

```diff
@@ -172,7 +172,9 @@ def get_app_and_build_ids(domain, build_or_app_id):
             copy_of = app_json.get('copy_of')
             if copy_of:
                 return copy_of, build_or_app_id
-    return build_or_app_id, None
+            return build_or_app_id, None
+        else:
+            return None, None
```

Comment:
Yeah, I'm not sure when this would happen or what we want to do when it does. The previous behavior just kind of "gave up" on doing something smart and returned what was passed in unmodified (thus emulating the behavior before we added the detect-app-from-build feature). Changing it to `None` would make the submission store `None` for app_id and build_id, which erases any trail of this happening at all. A more aggressive approach would be to raise an error here and 500 if this happens, and a toned down version of that would be to keep the previous behavior but add a soft-assert.

Do we know how/when this might happen? To me that informs what we'd want to do in that situation.